sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/HttpInstrumentationOptions.java

```diff
@@ -0,0 +1,299 @@
+     * for requests and responses where the content length is not known or greater than 16KB.
+     *
+     * @return True if content logging is enabled, false otherwise.
+     */
+    public boolean isContentLoggingEnabled() {
+        return isContentLoggingEnabled;
+    }
+
+    /**
+     * Enables or disables logging of HTTP request and response.
+     * False by default.
+     * <p>
+     * When HTTP logging is disabled, basic information about the request and response is still recorded
+     * via distributed tracing.
+     *
+     * @param isHttpLoggingEnabled True to enable detailed HTTP logging, false otherwise.
+     * @return The updated {@link HttpInstrumentationOptions} object.
+     */
+    public HttpInstrumentationOptions setHttpLoggingEnabled(boolean isHttpLoggingEnabled) {
+        this.isHttpLoggingEnabled = isHttpLoggingEnabled;
+        return this;
+    }
+
+    /**
+     * Enables or disables logging of HTTP request and response body. False by default.
+     * Enabling content logging also enables HTTP logging in general.
+     * <p>
+     * Note: even when content logging is explicitly enabled, content is not logged for requests and responses where the
+     * content length is not known or greater than 16KB.
+     *
+     * @param isContentLoggingEnabled True to enable content logging, false otherwise.
+     * @return The updated {@link HttpInstrumentationOptions} object.
+     */
+    public HttpInstrumentationOptions setContentLoggingEnabled(boolean isContentLoggingEnabled) {
+        this.isHttpLoggingEnabled |= isContentLoggingEnabled;
+        this.isContentLoggingEnabled = isContentLoggingEnabled;
+        return this;
+    }
+
+    /**
+     * Gets the allowed headers that should be logged when they appear on the request or response.
+     *
+     * @return The list of allowed headers.
+     */
+    public Set<HttpHeaderName> getAllowedHeaderNames() {
+        return Collections.unmodifiableSet(allowedHeaderNames);
+    }
+
+    /**
+     * Sets the given allowed headers that should be logged.
+     * Note: headers are not recorded on traces.
+     *
+     * <p>
+     * This method sets the provided header names to be the allowed header names which will be logged for all HTTP
+     * requests and responses, overwriting any previously configured headers. Additionally, users can use
+     * {@link HttpInstrumentationOptions#addAllowedHeaderName(HttpHeaderName)} or {@link HttpInstrumentationOptions#getAllowedHeaderNames()} to add or
+     * remove more headers names to the existing set of allowed header names.
+     * </p>
+     *
+     * @param allowedHeaderNames The list of allowed header names from the user.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpInstrumentationOptions setAllowedHeaderNames(final Set<HttpHeaderName> allowedHeaderNames) {
+        this.allowedHeaderNames = allowedHeaderNames == null ? new HashSet<>() : allowedHeaderNames;
```

Comment:
Should copy construct when `allowedHeaderNames` isn't null as this can run into weird edge cases if this is an unmodifiable Set. Same goes for query params.